### PR TITLE
#107 | Page tabs and routing fix

### LINKED ISSUE #107: Tabs don't navigate properly
## Overview

Routing to conditional tab should be handled.

## Steps to reproduce

1. Go to https://www.teloscan.io/address/0xfa9da51631268a30ec3ddd1ccbf46c65fad99251#contract
2. See the empty screen with no data and no indication that this is not supposed to exist

## Current behavior

The page does not redirect when direct-loading a conditionally valid tab, resulting in a blank screen

Image

## Expected behavior

Redirect the user to the Transactions tab

### PR DESCRIPTION
**Pull request by EJSG**
# Fixes #107

## Description
Route hashes, which are used by the quasar tabbing component, currently break if an invalid route is navigated to. This PR fixes how hashes are handled.

## Test scenarios

- go to https://deploy-preview-263--dev-mainnet-teloscan.netlify.app/address/0xdb45411265782debf29a7b58ab475bf75c9b19f3
    - the page should load fine with the Transaction tab selected
- go to https://deploy-preview-263--dev-mainnet-teloscan.netlify.app/address/0xdb45411265782debf29a7b58ab475bf75c9b19f3#transactions
    - the page should load fine with the Transaction tab selected
- on the same page, click around to the different tabs (ERC20, ERC721, etc)
    - tabs should load as normal
- go to an invalid hash, like https://deploy-preview-263--dev-mainnet-teloscan.netlify.app/address/0xdb45411265782debf29a7b58ab475bf75c9b19f3#fake
    - you should be directed to the transactions tab
- go to the `#contract` tab for an address which has no contract tab, such as https://deploy-preview-263--dev-mainnet-teloscan.netlify.app/address/0xdb45411265782debf29a7b58ab475bf75c9b19f3#contract
    - you should be directed to the transactions tab
- go to the `#contract` tab for an address which DOES have a contract tab, such as https://deploy-preview-263--dev-mainnet-teloscan.netlify.app/address/0xd102ce6a4db07d247fcc28f366a623df0938ca9e#contract
    - the contract tab should be loaded
- go to a transaction page, e.g. https://deploy-preview-263--teloscan-stage.netlify.app/tx/0xbd2510d29472bf0ec2ba9def6709e306a7c8f1e1fb1297e3e18279aa5ca3e77d#general
- click around to different tabs 
    - the tabs should load normally    
- go to https://deploy-preview-263--teloscan-stage.netlify.app/staking (you will need to be signed in for this)
- click around different tabs
    - tabs should load normally

## Checklist:

-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have cleaned up the code in the areas my change touches
-   [x] My changes generate no new warnings
-   [x] Any dependent changes have been merged and published in downstream modules
-   [x] I have checked my code and corrected any misspellings
-   [x] I have removed any unnecessary console messages

### CODE REVIEW

File: src/pages/staking/Staking.vue [hunk 1]
```diff
@@ -193,6 +199,24 @@ export default {
                 }
             },
         },
+        $route: {
```
Comment (donnyquixotic) [inline src/pages/staking/Staking.vue:202]: this is identical to the implementation in `Transaction.vue`, refactor if practical.

File: src/pages/Transaction.vue [hunk 2]
```diff
@@ -66,6 +73,24 @@ export default {
             },
             immediate: true,
         },
+        $route: {
```
Comment (donnyquixotic) [inline src/pages/Transaction.vue:76]: this is identical to the implementation in `Staking.vue`, refactor if practical.

**Review by donnyquixotic — CHANGES_REQUESTED** [review]

**Review by karynemayer — APPROVED** [review]
I'm good to merge it once the refactor is in place

**Comment by EJSG** [comment]
@donnyquixotic @karynemayer updated

**Review by karynemayer — APPROVED** [review]

**Review by donnyquixotic — DISMISSED** [review]

**Pull request merged**